lib/Dialect/FIRRTL/Transforms/EmitOMIR.cpp

```diff
@@ -1208,6 +1244,51 @@ hw::InnerRefAttr EmitOMIRPass::getInnerRefTo(FModuleLike module,
                          });
 }
 
+FIRRTLType EmitOMIRPass::getTypeOf(Operation *op) {
+  assert(op->getNumResults() == 1 &&
+         op->getResult(0).getType().isa<FIRRTLType>() &&
+         "op must have a single FIRRTLType result");
+  return op->getResult(0).getType().cast<FIRRTLType>();
+}
+
+FIRRTLType EmitOMIRPass::getTypeOf(FModuleLike mod, size_t portIdx) {
+  Type portType = mod.getPortType(portIdx);
+  assert(portType.isa<FIRRTLType>() && "port must have a FIRRTLType");
+  return portType.cast<FIRRTLType>();
+}
```

Comment:
Out-of-scope: It seems weird that an `FModuleLike` returns a port of `Type` as opposed to `FIRRTLType`. Maybe we should change this.

Reply:
I think it's just lifting a TypeAttr and returning the Type from it, should be safe to cast that to FIRTTLType. I'll open an issue.